fix:Add toast message when no user with common skills is found using filter(anita-org#866)

ritvij14 opened the pull request:
### Description
Added Toast for the situation when members with common skills are not found.

Fixes #866 

### Type of Change:
- Code

**Code/Quality Assurance Only**
- Bug-fix (non-breaking change which fixes an issue)


### How Has This Been Tested?
I tested it manually on my device Redmi Note 8 Pro. [Video](https://drive.google.com/file/d/1DpQejuCuRyxjTBaqpKQYCqrMQFwh4GoZ/view?usp=sharing)


### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] Any dependent changes have been merged


**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

ritvij14 committed:
fix:Add toast message when no user with common skills is found using filter(anita-org#866)

spacemonkey1101 commented:
is this issue open?

mtreacy002 reviewed (changes requested):
@ritvij14 , I've tested this PR locally on a Pixel 2 emulator. I found a bug in the code that the toast `Ooops! Not found` pop up not only when the filter is applied, but even before that (when logged-in user select `Members` icon from the bottom Navbar). Can you please fix this? Thanks

![ezgif com-gif-maker (8)](https://user-images.githubusercontent.com/29667122/96080978-0ac68400-0f04-11eb-9585-0f771fbd9282.gif)

cc @anna4j  and @sanchi0204

ritvij14 commented:
@mtreacy002 sorry for the late response :sweat_smile: 

Could you please guide me as to where should I add the toast to not face this bug? I am really confused.

mtreacy002 commented:
@ritvij14 , when you run the app with the debugger, you can see that on its initial state when user first click the `Members` page (providing the app is newly installed, to avoid getting side effect from cache of previous actions), the filterMap on line 119 of `MembersFragment` will show only '"sort_key" -> "REGISTRATION_DATE"` as the filter condition.
<img width="1468" alt="Screen Shot 2020-10-18 at 4 38 46 pm" src="https://user-images.githubusercontent.com/29667122/96359751-1a430880-1162-11eb-89d2-7dab0909c11d.png">

Then when you type in a specific filter, the filterMap will show 3 other attributes with "skills" showing the string you just typed (in my example its "python").
<img width="1475" alt="Screen Shot 2020-10-18 at 4 40 08 pm" src="https://user-images.githubusercontent.com/29667122/96359753-23cc7080-1162-11eb-8b0a-f9d953854fdf.png">

Even if user clear all filters, the filterMap will show all 4 attributes (sort_key + 3 filters) but with filters getting empty strings.
<img width="1472" alt="Screen Shot 2020-10-18 at 4 46 43 pm" src="https://user-images.githubusercontent.com/29667122/96359775-668e4880-1162-11eb-8811-1a7d7c08a2b7.png">


You can use this information to place an if condition to check the status of filterMap and apply toast only when the desired filter exist in filterMap and not an empty string.

Hope this helps

ritvij14 commented:
@mtreacy002 thanks for the clarification, I will add an if condition for the filters. But this bug didn't happen on my device, is that because of different data in the recycler view or something else?


![20201018_124650](https://user-images.githubusercontent.com/51456744/96361155-f28f7880-1140-11eb-80e5-f719721d68f1.gif)

ritvij14 committed:
checking if skills filter is applied or not

mtreacy002 reviewed (changes requested):
@ritvij14 . I tested the latest change but the toast ended up not showing at all, both on initial click to `Members` page and after the filter is applied.
* tested with `debug_localhost` as active build variant
![pr-884](https://user-images.githubusercontent.com/29667122/97058209-dc6a3800-15d8-11eb-9d8f-ca3435343171.gif)

* tested with `debug` as active build variant
![pr-884b](https://user-images.githubusercontent.com/29667122/97058619-0cfea180-15da-11eb-95eb-1ad2f0840526.gif)

Can you please fix this?

ritvij14 committed:
Merge branch 'develop' of https://github.com/anitab-org/mentorship-android into skill-toast-fix

ritvij14 committed:
modified condition

ritvij14 commented:
@mtreacy002 yeah sorry about that 😅 I didn't notice that I had placed the wrong condition there. I have changed it and this is the current look. Does it look ok now?

![20201024_123456 1](https://user-images.githubusercontent.com/51456744/97070393-9cfe1480-15f5-11eb-994c-262eb46a67ee.gif)

Hyper-Matrix committed:
feat: confirmation dialog when accepting / rejecting / deleting request (#949)

Hyper-Matrix committed:
fix: Keyboard appearance in the Signup page (#895)

priyanshuone6 committed:
fix: disorientation of the cancel button in landscape mode (#889)

ankit-saraf committed:
fix: filter users by non case sensitive inputs (#875)

CodeChamp-SS committed:
fix: centering the ui in relations tab (#819)

siddheshkothadi committed:
fix: disabled the login button until the user has filled all the credentials (#812)

DhaneshShetty committed:
fix: maladaptive about page constraints (#798)

justdvnsh committed:
tests: additional tests for signup feature (#936)

justdvnsh committed:
tests: fixed all the failing Login Activity Tests (#921)

body: Fixed the issue for #918. Fixed all the failing tests in the login activity

Anmollenka committed:
docs: add to README.md instructions on how to run UI tests (#888)

siddheshkothadi committed:
chore: add GitHub Actions that run same steps as Travis CI (#876)

Co-authored-by: Isabel Costa <11148726+isabelcosta@users.noreply.github.com>

CodeChamp-SS committed:
fix: feedback screen validation error (#991)

atharwa-24 committed:
docs: add logo to the existing badges on README (#998)

Added logos to get the info. about the repos ci ,etc system at first glance

justdvnsh committed:
test: add Logout test (#834)

Add Test For the Logout Button in Settings Activity to check whether the button is logging the user out. Fix for Issue #801

TreavVasu committed:
style: change App name to Mentorship System (#814)

justdvnsh committed:
test: add test for Change Password Fragment (#836)

* test: Added Test for Change Password Fragment - Scenario (b)
body: Added Test for the Change Password -> When the password does not match. Fixes #804

somayaj committed:
style: remove redundant app namespace (#1032)

* Removed redundant app namespace - fix for #1008
* Removed redundant android namespace - fix for #1008

epicadk committed:
fix: replace Systers with AnitaB.org in the About screen (#961)

shivamparashar165 committed:
fix: spelling mistake in SignUpActivityTest (#1045)

ankit-saraf committed:
docs: audit accessibility issues (#932)

priyanka0906 committed:
docs: fix unit test mention to UI in README (#1043)

spacemonkey1101 committed:
fix: mentorship system logo systers mention (#1012)

DhaneshShetty committed:
fix: resolve long text cut in bottom nav (#853)

DhaneshShetty committed:
feat: set condition for enabling change password btn (#837)

faznan3nazer committed:
docs: add feature list to Readme.md (#966)

maharishi-01 committed:
docs: create test workflow for reject mentorship request (#964)

Sonali-Sharma-1 committed:
feat: LoginActivity has singleTask launchMode attribute (#1037)

Rahulm2310 committed:
docs: add kotlin docs link in readme.md and contributing.md (#1067)

Sanika1999 committed:
docs: comment out description placeholder in templates (#1068)

* docs : Comment out descriptions

* Update bug_report.md

* Update PULL_REQUEST_TEMPLATE.md

devsebastian committed:
test: Request Detail Screen Espresso test (a) (#1060)

devsebastian committed:
test: "Request Detail" Screen Espresso test (b) (#1059)

darahask committed:
feat: runtime checks to signup for the email and confirm password (#954)

This adds runtime validation for the email and confirm password.
This resolves 2nd and 3rd point of issue#782.

swaraj961 committed:
docs: updated the code of conduct (#885)

Zulfa210 committed:
chore: use anitab instead of systers in the app id (#1071)

Changed application id from org.systers.mentorship to org.anitab.mentorship

varunvjha committed:
docs: update Code of Conduct (#1078)

Preshit-Joshi committed:
style: fix hardcoded strings in SignUp activity (#1076)

werainkhatri committed:
docs: guideline to resize image and table in markdown (#1082)

kartikeysaran committed:
chore: delete .travis.yml (#1084)

HeroicHitesh committed:
docs: manual test for create tasks and complete the task (#906)

Co-authored-by: Isabel Costa <11148726+isabelcosta@users.noreply.github.com>

epicadk committed:
test: fix login activity tests (#1030)

epicadk committed:
test: fix bug when user enters incorrect password (#984)

CodeChamp-SS reviewed (dismissed):
LGTM ! Tested locally on emulator as well. Thanks @ritvij14 for your contribution.

vj-codes commented:
@CodeChamp-SS can you please add a gif if you've tested on emulator just to crosscheck it's functionality, thanks!

AkinchanKushwaha committed:
docs: delete .github/ISSUE_TEMPLATE folder (#1105)

CodeChamp-SS commented:
> @CodeChamp-SS can you please add a gif if you've tested on emulator just to crosscheck it's functionality, thanks!

on testing in different scenarios, I found a bug 😅. Even when the result is found, the toast pops up.

@ritvij14 can you please fix it ? I'll be happy to test again once you've fixed this issue :-)

https://user-images.githubusercontent.com/52480794/110072233-79b55400-7da3-11eb-9f68-89a03bcc7d18.mp4

ritvij14 commented:
@CodeChamp-SS yes ok I will have a look once again and get back to you, will try to correct the errors asap.

vj-codes commented:
@CodeChamp-SS thanks for uploading it :)

ritvij14 committed:
Merge branch 'skill-toast-fix' of github.com:ritvij14/mentorship-android into skill-toast-fix

ritvij14 commented:
@CodeChamp-SS in the most recent pull of the latest code, my Gradle build is getting stopped and failing showing this error:- 
`Gradle sync failed: Gradle build daemon disappeared unexpectedly`

This is not happening in other android projects on my system.

CodeChamp-SS commented:
> @CodeChamp-SS in the most recent pull of the latest code, my Gradle build is getting stopped and failing showing this error:-
> `Gradle sync failed: Gradle build daemon disappeared unexpectedly`
> 
> This is not happening in other android projects on my system.

do one thing revert all the commits from your branch, then pull the develop branch, then commit your changes. Then try to build, ig that should fix the problem. If your changes are working as expected then do force push on the branch

ritvij14 commented:
> > @CodeChamp-SS in the most recent pull of the latest code, my Gradle build is getting stopped and failing showing this error:-
> > `Gradle sync failed: Gradle build daemon disappeared unexpectedly`
> > This is not happening in other android projects on my system.
> 
> do one thing revert all the commits from your branch, then pull the develop branch, then commit your changes. Then try to build, ig that should fix the problem. If your changes are working as expected then do force push on the branch

Would that fix the issue though? Because I have this on my local copy's `develop` branch too and it is completely even with the current develop branch.

CodeChamp-SS commented:
> > > @CodeChamp-SS in the most recent pull of the latest code, my Gradle build is getting stopped and failing showing this error:-
> > > `Gradle sync failed: Gradle build daemon disappeared unexpectedly`
> > > This is not happening in other android projects on my system.
> > 
> > 
> > do one thing revert all the commits from your branch, then pull the develop branch, then commit your changes. Then try to build, ig that should fix the problem. If your changes are working as expected then do force push on the branch
> 
> Would that fix the issue though? Because I have this on my local copy's `develop` branch too and it is completely even with the current develop branch.

hmm then ig you can try cleaning and then rebuilding the project, if the problem still exists i think you'll have to delete the project and then clone it again in your machine, this the only solution i could think of 😅 as i've never faced such an issue

CodeChamp-SS commented:
you may also try asking on zulip may be some1else could be able to help you :-)
